pkg/gui/filetree/node.go

```diff
@@ -155,6 +155,22 @@ func (self *Node[T]) EveryFile(test func(*T) bool) bool {
 	return true
 }
 
+func (self *Node[T]) FindFirstFile(test func(*T) bool) *T {
```

Comment:
I found the name of this function a bit confusing at first, until I realized that it means "find first file that matches the given predicate". I wonder if there's some way to make this more obvious. Maybe appending `By` is already enough? This seems to be a convention used by samber/lo, e.g. [lo.FindKeyBy](https://github.com/jesseduffield/lazygit/blob/facc73a88b72a0739563017e6d6e7bf0a41f3366/vendor/github.com/samber/lo/find.go#L101).

I'd also rename `test` to `predicate`, but I see there's already precedence for the name `test` in this file. (Though I wouldn't mind a separate cleanup commit that renames them all to `predicate`, but maybe that's just personal taste. I'll leave that to you to decide.)